### PR TITLE
55 read children by id api endpoint

### LINKED ISSUE #55: Read-children-by-id API endpoint
### Description
For our dynamic loading strategy, the frontend will fetch new nodes from the backend when a node is expanded. Create an endpoint to serve the children of a given node.

Might need some discussion - I was thinking that if the called node is "Thing", you would also just serve "Thing" itself (it's our root node). A separate endpoint seemed like overkill but let Heidi know if it overcomplicates things.

### To-do
* [ ] New endpoint with id query parameter
* [ ] Get a single node's direct children from the database
* [ ] Serve the direct children with their id, labels and deprecation date if applicable
* [ ] Deal with sending back "Thing"

### PR DESCRIPTION
**Pull request by keyman015**
## Description
I have completed the API endpoint for the basic direct children and root node. The routes are `/graph/children?id=<id>` and `/graph/root` (both using GET method). By default, deprecated nodes WILL NOT be returned, however they can be included with argument `dep=true` in the request. Where it will return in such a way:

image

When getting the children, their label, id and deprecation date (if they dont have one its `null`) are returned in json format.

image

To accompany these changes to backend server, a bunch of new tests were written to ensure complete code coverage.

## To-do List
* [x] New endpoint with id query parameter
* [x] Get a single node's direct children from the database
* [x] Serve the direct children with their id, labels and deprecation date if applicable
* [x] Deal with sending back "Thing"

## Misc
Closes #55

### CODE REVIEW

**Review by paulmmaingi — COMMENTED** [review]
Nice! Looks good.

The only other suggestion I have would be adding docstrings to the python functions to keep it consistent.

**Comment by keyman015** [comment]
image

Extra parents have been implemented into the `/graph/children` with the field `extra_parents=true/false`, by default it returns the additional parents

File: src/server/app/routes.py [hunk 1]
```diff
@@ -13,3 +14,56 @@ def home():
...
+        return jsonify(root_node_info)
+
+    else:
+        return jsonify({"error": "No ROOT found"}), 404
+
+
+@main.route("/graph/children", methods=["GET"])
+def get_children():
+    # Extract the custom ID from the query parameters
+    node_uri = request.args.get("id")
```
[the remark this answers is not in the document]
Comment (keyman015) [reply src/server/app/routes.py]: The path has also now been changed to `/graph/children/<path:node_uri>` to match the standards.

**Review by paulmmaingi — APPROVED** [review]
LGTM

**Pull request merged**